src/sc/sidechainTxsCommitmentBuilder.cpp

```diff
@@ -0,0 +1,177 @@
+
+    // set of scid is ordered
+    for (const auto& scid : sScIds)
+    {
+        field_t* ftRootField = nullptr;
+        auto itFt = mScMerkleTreeLeavesFt.find(scid);
+        if (itFt != mScMerkleTreeLeavesFt.end() )
+        {
+            ftRootField = merkleTreeRootOf(itFt->second);
+        }
+
+        field_t* btrRootField = nullptr;
+        auto itBtr = mScMerkleTreeLeavesBtr.find(scid);
+        if (itBtr != mScMerkleTreeLeavesBtr.end() )
+        {
+            btrRootField = merkleTreeRootOf(itBtr->second);
+        }
+
+        field_t* certRootField = nullptr;
+        auto itCert = mScCerts.find(scid);
+        if (itCert != mScCerts.end() )
+        {
+            certRootField = itCert->second;
+        }
+
+        field_t* scIdField = mapCertToField(scid);
+
+        std::vector<field_t*> singleSidechainComponents = {ftRootField, btrRootField, certRootField, scIdField};
+        field_t * sidechainTreeRoot = merkleTreeRootOf(singleSidechainComponents);
+
+        vSortedScLeaves.push_back(sidechainTreeRoot);
+    }
+
+    field_t * finalTreeRoot = emptyField.zero;
+    if (vSortedScLeaves.size() != 0)
+        finalTreeRoot = merkleTreeRootOf(vSortedScLeaves);
+
+    uint256 res = mapFieldToHash(finalTreeRoot);
+    if (finalTreeRoot != emptyField.zero) zendoo_field_free(finalTreeRoot);
+
+    return res;
+}
+#endif
+
+field_t* SidechainTxsCommitmentBuilder::mapScTxToField(const uint256& ccoutHash, const uint256& txHash, unsigned int outPos) const
+{
+    static_assert((sizeof(uint256) + sizeof(uint256) + sizeof(unsigned int)) < SC_FIELD_SIZE);
+
+    std::string toHash = std::string(ccoutHash.begin(), ccoutHash.end()) +
+            std::string(txHash.begin(), txHash.end()) +
+            std::string(static_cast<char*>(static_cast<void*>(&outPos)));
+
+    unsigned char hash[SC_FIELD_SIZE] = {};
+    memcpy(hash, toHash.append(SC_FIELD_SIZE - toHash.size(), '\0').c_str(), SC_FIELD_SIZE);
+
+    //generate field element. MISSING null-check
+    field_t* field = zendoo_deserialize_field(hash);
+    return field;
+}
+
+field_t* SidechainTxsCommitmentBuilder::mapCertToField(const uint256& certHash) const
+{
+    static_assert(sizeof(uint256) < SC_FIELD_SIZE);
+    unsigned char hash[SC_FIELD_SIZE] = {};
+    std::string resHex = std::string(certHash.begin(), certHash.end());
+    memcpy(hash, resHex.append(SC_FIELD_SIZE - resHex.size(), '\0').c_str(), SC_FIELD_SIZE);
+
+    //generate field element. MISSING null-check
+    field_t* field = zendoo_deserialize_field(hash);
+    return field;
+}
+
+uint256 SidechainTxsCommitmentBuilder::mapFieldToHash(const field_t* pField) const
+{
+    if (pField == nullptr)
+        throw;
+
+    unsigned char field_bytes[SC_FIELD_SIZE];
+    zendoo_serialize_field(pField, field_bytes);
+
+    uint256 res = Hash(BEGIN(field_bytes), END(field_bytes));
+    return res;
+}
+
+unsigned int SidechainTxsCommitmentBuilder::treeHeightForLeaves(unsigned int numberOfLeaves) const
```

Comment:
It's ok for a temporary implementation, but in the SNARKs we need to have constant height for all the Merkle Trees, so this function won't be needed.